client/rest-high-level/src/main/java/org/elasticsearch/client/FreezeIndexRequest.java

```diff
@@ -18,27 +18,79 @@
  */
 package org.elasticsearch.client;
 
+import org.elasticsearch.action.admin.indices.open.OpenIndexResponse;
+import org.elasticsearch.action.support.ActiveShardCount;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.client.core.AcknowledgedRequest;
+
 import java.util.Objects;
 
 /**
  * Request for the _freeze index API
  */
-public final class FreezeIndexRequest implements Validatable {
+public final class FreezeIndexRequest extends AcknowledgedRequest {
 
-    private final String indices;
+    private final String[] indices;
+    private IndicesOptions indicesOptions;
+    private ActiveShardCount waitForActiveShards;
 
     /**
      * Creates a new freeze index request
      * @param indices the index to freeze
      */
-    public FreezeIndexRequest(String indices) {
+    public FreezeIndexRequest(String... indices) {
         this.indices = Objects.requireNonNull(indices);
     }
 
     /**
      * Returns the indices to freeze
      */
-    public String getIndices() {
+    public String[] getIndices() {
```

Comment:
Id much prefer to see us use a List here, even tho we take in a variadic array. No reason to have users be using a array if they can get by with a List. Maybe convert it in the constructor.

Reply:
RequestConverters uses arrays all over the place. It's the main consumer. Why would we change that?

Reply:
thats a fair point.